content/blog/2023/07/10/2023-07-12-jenkins-mirrors-postmortem-outage.adoc

```diff
@@ -0,0 +1,119 @@
+** ci.jenkins.io
+** infra.ci.jenkins.io
+** issues.jenkins.io
+** plugins.jenkins.io
+** repo.jenkins-ci.org
+** www.jenkins.io
+
+== Incident Timeline
+
+* **10:30am UTC:** after a successful upgrade of the public Kubernetes cluster in Azure to 1.25 (as part of https://github.com/jenkins-infra/helpdesk/issues/3582[]), we realized that the LDAP service was not reachable by the services running inside the cluster (such as accounts.jenkins.io).
+We quickly realize that the IP restrictions were blocking these requests as the pod originating IP was in a different range than i t was before.
+
+* **10:55am UTC:** The fix https://github.com/jenkins-infra/azure/pull/431 is deployed to specify a proper set of IP ranges for the pods. It removed all of the node pools (e.g. all the virtual machines where the container were running) and failed to re-create them causing a full outage of all the services running in this cluster:
+** accounts.jenkins.io
+** get.jenkins.io
+** incrementals.jenkins.io
+** javadoc.jenkins.io
+** jenkins-wiki-exporter.jenkins.io
+** ldap.jenkins.io
+** plugins.jenkins.io
+** previews of *.jenkins.io pull requests (infra.ci.jenkins.io)
+** release.ci.jenkins.io
+** repo.azure.jenkins.io
+** reports.jenkins.io
+** stories.jenkins.io
+** uplink.jenkins.io
+** www.jenkins.io
+
+* **11:16am until 13:16pm UTC:** The failure to re-create resources led us to spend the 2 next hours in creating the cluster from scratch with a fixed network setup.
+
+* **15:17pm UTC:** https://github.com/jenkins-infra/azure/pull/432 is pushed to persist the manual work we did to recreate the cluster including the IP setup.
+
+* **15:25pm UTC:** all services are back to normal
+
+== What Happened?
+
+* When the cluster was initially created, we selected the virtual network's IP range `10.**244**.0.0/**16**` (ref. https://github.com/jenkins-infra/azure-net/blob/fcb010a5d9f164203c9a896fcb974df4051c321d/vnets.tf#L66) with a sub-network `10.245.0.0/24` (ref. https://github.com/jenkins-infra/azure-net/blob/fcb010a5d9f164203c9a896fcb974df4051c321d/vnets.tf#L161)
+
+* But we ignored that the range `10.**244**.0.0/**24**` is the default CIDR for the Kubernetes Pods network in Azure when using the link:https://learn.microsoft.com/en-us/azure/aks/configure-kubenet["kubenet" network to support IPv6 instead of default CNI]
+
+* The node pool re-creation failed because we assumed both ranges were able to communicate and tried to deploy and invalid setup.
+
+* As soon as we specified a custom Pod CIDR in a distinct range, everything went fine
+
+* Also, when the original cluster was deleted, it transitively removed the current Public IPs as it removed the link:https://learn.microsoft.com/en-us/azure/aks/faq#why-are-two-resource-groups-created-with-aks[Nodes Resource Group] containing the Public IP
+** These public IPs should change as less as possible to avoid problems with our users running behind a corporate firewall with an allow-list
```

Comment:
```suggestion
** These public IPs should change as little as possible to avoid problems with our users running behind a corporate firewall with an allow-list.
```
suggestion for syntax